src/joystick/joystick.cpp

```diff
@@ -0,0 +1,74 @@
+#include <string>
+#include <cmath>
+#include <ctime>
+#include <boost/optional.hpp>
+#include <math.h>
+#include <map>
+
+#include "joystick_manager.h"
+
+int JoyEntry::intSupplier = 0;                  // Set first joystick id to 0
```

Comment:
what does the intSupplier do?

Reply:
It's pretty much a count that assigns the next number to a new incoming controller

Reply:
Should it be changed, or better documented. I thought the small comment next to it should do